### PR TITLE
Remove HACK that used to put `Handle` in config bag

### PR DESCRIPTION
**Pull request by ysaito1001**
## Motivation and Context
Removes `HACK` that used to put `Handle` in config bag.

## Description
The `HACK` was previously placed in `ServiceRuntimePlugin::config` because later in the orchestrator execution, we needed to access service config fields (through `Handle`) to build endpoint `Params` within `EndpointParamsInterceptor`. Now that service config fields are baked into a frozen layer, `EndpointParamsInterceptor` can load those fields directly from the config bag (to which the frozen layer has been added) when constructing endpoint `Params`. We can therefore remove `HACK` at this point.

## Testing
- [x] Passed tests in CI

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Review by jdisanti — APPROVED** [review]
Nice work!

File: codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/config/ServiceConfigGenerator.kt [hunk 1]
```diff
@@ -190,31 +213,6 @@ fun standardConfigParam(param: ConfigParam, codegenContext: ClientCodegenContext
                 }
             }
 
-            ServiceConfig.ConfigImpl -> writable {
```
[the remark this answers is not in the document]
Comment (ysaito1001) [reply codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/config/ServiceConfigGenerator.kt:193]: What's been removed here are `pub(crate)` getters so the customers could've used it in the first place 🙂 
These are `standardConfigParam`s, and we still have the other `pub` getters on service config defined by other customizations.

File: codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/ServiceRuntimePluginGenerator.kt [hunk 2]
```diff
@@ -111,6 +111,9 @@ class ServiceRuntimePluginGenerator(
     fun render(writer: RustWriter, customizations: List<ServiceRuntimePluginCustomization>) {
         writer.rustTemplate(
             """
+            // TODO(enableNewSmithyRuntimeLaunch) Remove `allow(dead_code)` as well as a field `handle` when
```
[the remark this answers is not in the document]
Comment (ysaito1001) [reply codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/ServiceRuntimePluginGenerator.kt:114]: Thanks for helping clarify this. I should actually update the comment. If we removed the field `handle`, we would see who's still referring to `self.handle.conf`:
```
         cfg.set_identity_resolvers(self.handle.conf.identity_resolvers().clone());
                                         ^^^^^^ unknown field
```
I expect we can soon address this in a subsequent PR.

[the remark this answers is not in the document]
Comment (jdisanti) [reply codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/ServiceRuntimePluginGenerator.kt:114]: Ah, it's the part I'm refactoring. OK. Sounds good.

**Pull request merged**